file_organizer_v2/src/file_organizer/services/deduplication/backup.py

```
@@ -0,0 +1,302 @@
+            List of paths to removed backup files
+        """
+        if max_age_days < 0:
+            raise ValueError("max_age_days must be non-negative")
+
+        cutoff_date = datetime.now() - timedelta(days=max_age_days)
+        manifest = self._load_manifest()
+        removed_backups = []
+
+        # Find and remove old backups
+        for backup_key, _metadata in list(manifest.items()):
+            backup_time = datetime.fromisoformat(metadata["backup_time"])
```

Ruff (0.14.13)

Undefined name `metadata`

(F821)